core/src/main/scala/com/stripe/agate/tensor/Tensor.scala

```diff
@@ -112,19 +114,22 @@ abstract class Tensor[D <: DataType] {
   def unsqueeze(axis: Long): Option[Tensor[dataType.type]] =
     dims.unsqueeze(axis).map(withDims)
 
-  def closeTo(that: Tensor[dataType.type], error: Float = 1e-5f): Boolean = {
-    val s0 = axes
-    val s1 = that.axes
-    val num = OnnxNumber.forDataType(dataType)
-    s0 == s1 && s0.forall { cs =>
-      val x0 = this(cs)
-      val x1 = that(cs)
-      Util.closeTo(num.toFloat(x0), num.toFloat(x1), error)
+  def closeTo(that: Tensor[dataType.type], error: Float = 1e-5f): Boolean =
+    OnnxNumber.forDataType(dataType) match {
+      case Success(num) =>
+        val s0 = axes
+        val s1 = that.axes
+        s0 == s1 && s0.forall { cs =>
+          val x0 = this(cs)
+          val x1 = that(cs)
+          Util.closeTo(num.toFloat(x0), num.toFloat(x1), error)
+        }
+      case _ =>
+        this == that
```

Comment:
If I am reading this correctly, for all non-numeric data types we are checking strict equality here? 

If that is a case, can you add a small comment above: 
```suggestion
       // For all non-numeric datatypes, perform a strict equality check
        this == that
```